src/github.com/matrix-org/dendrite/roomserver/query/query.go

```diff
@@ -709,3 +710,130 @@ func (r *RoomserverQueryAPI) SetupHTTP(servMux *http.ServeMux) {
 		}),
 	)
 }
+
+// InProcessRoomServerQueryAPI is a RoomserverQueryAPI database linked with a
+// OpenTracing tracer, providing metrics
+type InProcessRoomServerQueryAPI struct {
+	db     RoomserverQueryAPI
+	tracer opentracing.Tracer
+}
+
+// NewInProcessRoomServerQueryAPI creates a new InProcessRoomServerQueryAPI
```

Comment:
Hm, not sure if this is entirely relevant to this PR, but was a part of the other branches. Thoughts @erikjohnston?

Reply:
Yes this is. It's so that we can trace calls between different components in the same process, like we do for the InternalAPI stuff